one_fm/api/mobile/shift_permission.py

```diff
@@ -31,17 +31,17 @@ def create_shift_permission(employee, permission_type, date, reason, leaving_tim
                 doc.shift_type = type
                 doc.save()
                 frappe.db.commit()
-                return response({'message':"Shift Permission Successfully Created",'data':doc},201)
+                return response("Shift Permission Successfully Created",doc,True,201)
             elif has_duplicate:
                 doc = frappe.get_doc('Shift Permission',{"employee": employee, "date":date, "assigned_shift": assigned_shift, "permission_type": permission_type})
-                return response({'message':"{0} has already applied for permission to {1} on {2}.".format(doc.emp_name, permission_type.lower(), date),'data':{}},409)
+                return response("{0} has already applied for permission to {1} on {2}.".format(doc.emp_name, permission_type.lower(), date), {}, False, 409)
 
         elif not shift or not type or not assigned_shift or not shift_supervisor:
-            return response({'message':"Shift Details are missing, Please Make sure You are have a Scheduled record on {0}".format(date),'data':{}},400)
+            return response("Shift Permission Not Created Successfully", {}, False, 400)
```

Comment:
Maybe change the wording? Something like "Could not create Shift Permission".

Reply:
Also, provide appropriate reason as to why Shift permission was not able to be created.